pkg/installation/utils.go

```diff
@@ -0,0 +1,232 @@
+	return
+}
+
+func (i *Installation) printInstallationErrorLog() error {
+	logs, err := i.getKubectl().RunCmd("get", "installation", "kyma-installation", "-o", "go-template", "--template={{- range .status.errorLog -}}{{printf \"%s:\n %s [%s]\n\" .component .log .occurrences}}{{- end}}")
+	if err != nil {
+		return err
+	}
+	fmt.Println(logs)
+	return nil
+}
+
+func (i *Installation) getMasterHash() (string, error) {
+	ctx, timeoutF := context.WithTimeout(context.Background(), 1*time.Minute)
+	defer timeoutF()
+	r, err := git.CloneContext(ctx, memory.NewStorage(), nil,
+		&git.CloneOptions{
+			Depth: 1,
+			URL:   "https://github.com/kyma-project/kyma",
+		})
+	if err != nil {
+		return "", err
+	}
+	h, err := r.Head()
+	if err != nil {
+		return "", err
+	}
+
+	return h.Hash().String()[:8], nil
+}
+
+func (i *Installation) setAdminPassword() error {
+	if i.Options.Password == "" {
+		return nil
+	}
+	encPass := base64.StdEncoding.EncodeToString([]byte(i.Options.Password))
+	_, err := i.k8s.Static().CoreV1().ConfigMaps("kyma-installer").Patch("installation-config-overrides", types.JSONPatchType,
+		[]byte(fmt.Sprintf("[{\"op\": \"replace\", \"path\": \"/data/global.adminPassword\", \"value\": \"%s\"}]", encPass)))
+	if err != nil {
+		err = errors.Wrap(err, "Error setting admin password")
+	}
+	return err
+}
+
+func removeActionLabel(acc *[]map[string]interface{}) error {
+	for _, config := range *acc {
+		kind, ok := config["kind"]
+		if !ok {
+			continue
+		}
+
+		if kind != "Installation" {
+			continue
+		}
+
+		meta, ok := config["metadata"].(map[interface{}]interface{})
+		if !ok {
+			return errors.New("Installation contains no METADATA section")
+		}
+
+		labels, ok := meta["labels"].(map[interface{}]interface{})
+		if !ok {
+			return errors.New("Installation contains no LABELS section")
+		}
+
+		_, ok = labels["action"].(string)
+		if !ok {
+			return nil
+		}
+
+		delete(labels, "action")
+
+	}
+	return nil
+}
+
+func buildDockerImageString(template string, version string) string {
+	return fmt.Sprintf(template, version)
+}
+
+func downloadFile(path string) (io.ReadCloser, error) {
+	client := &http.Client{
+		Timeout: 5 * time.Second,
+	}
+	resp, err := client.Get(path)
+	if err != nil {
+		return nil, err
+	}
+	return resp.Body, nil
+}
+
+func getInstallerImage(resources *[]map[string]interface{}) (string, error) {
+	for _, res := range *resources {
+		if res["kind"] == "Deployment" {
+
+			var deployment v1.Deployment
+			err := mapstructure.Decode(res, &deployment)
+			if err != nil {
+				return "", err
+			}
+
+			if deployment.Spec.Template.Spec.Containers[0].Name == "kyma-installer-container" {
+				return deployment.Spec.Template.Spec.Containers[0].Image, nil
+			}
+		}
+	}
+	return "", errors.New("'kyma-installer' deployment is missing")
+}
+
+func replaceInstallerImage(resources *[]map[string]interface{}, imageURL string) error {
+	for _, config := range *resources {
+		kind, ok := config["kind"]
```

Comment:
Maybe we can reduce the length and improve readability of this function by using positive if statements instead of negative ones + continue.
Something like:
```go
for _, config := range *resources {
    if kind, ok := config["kind"]; ok && kind == "deployment" {
        if spec, ok := config["spec"].(map[interface{}]interface{}); ok {
            if template, ok := spec["template"].(map[interface{}]interface{}); ok {
                ....
            }
        }
    }
}
```

The downside of this is a lot of nested if statements, not sure which is better. 🤔